ui/pages/Pool.tsx

```diff
@@ -0,0 +1,140 @@
+
+  const { data, isPlaceholderData, isError, error } = useApiQuery('pool', {
+    pathParams: { hash, chainId: config.chain.id },
+    queryOptions: {
+      placeholderData: POOL,
+      refetchOnMount: false,
+    },
+  });
+
+  const addressQuery = useApiQuery('address', {
+    pathParams: { hash: data?.contract_address },
+    queryOptions: {
+      enabled: Boolean(data?.contract_address),
+      placeholderData: addressStubs.ADDRESS_INFO,
+    },
+  });
+
+  const content = (() => {
+    if (isError) {
+      if (isCustomAppError(error)) {
+        throwOnResourceLoadError({ resource: 'pool', error, isError: true });
+      }
+
+      return <DataFetchAlert/>;
+    }
+
+    if (!data) {
+      return null;
+    }
+
+    return (
+      <PoolInfo
+        data={ data }
+        isPlaceholderData={ isPlaceholderData }
+      />
+    );
+  })();
+
+  // More links can be added here
+  const externalLinks = [ {
+    url: data?.coin_gecko_terminal_url,
+    image: '/static/gecko_terminal.png',
+    title: 'GeckoTerminal',
+  } ].filter(link => Boolean(link.url));
+  const hasLinks = externalLinks.length > 0;
+
+  const externalLinksComponents = React.useMemo(() => {
+    return externalLinks
+      .map((link) => {
+        return (
+          <LinkExternal h="34px" key={ link.url } href={ link.url } alignItems="center" display="inline-flex" minW="120px">
+            <Image boxSize={ 5 } mr={ 2 } src={ link.image } alt={ `${ link.title } icon` }/>
+            { link.title }
+          </LinkExternal>
+        );
+      });
+  }, [ externalLinks ]);
+
+  const titleSecondRow = (
+    <Flex alignItems="center" justifyContent="space-between" w="100%">
+      { addressQuery.data ? <AddressEntity address={ addressQuery.data } isLoading={ addressQuery.isPlaceholderData }/> : <Box/> }
+      <Flex gap={ 2 }>
+        <InfoButton>
+          { `This Liquidity Provider (LP) token represents ${ data?.base_token_symbol }/${ data?.quote_token_symbol } pairing.` }
+        </InfoButton>
+        { hasLinks && (
+          <VerifyWith
+            links={ externalLinksComponents }
+            label=""
+            longText="View in"
+            shortText=""
+          />
+        ) }
+      </Flex>
+    </Flex>
+  );
+
+  const backLink = React.useMemo(() => {
+    const hasGoBackLink = appProps.referrer && appProps.referrer.includes('/pools');
+
+    if (!hasGoBackLink) {
+      return;
+    }
+
+    return {
+      label: 'Back to pools list',
+      url: appProps.referrer,
+    };
+  }, [ appProps.referrer ]);
+
+  const poolTitle = data ? `${ data.base_token_symbol } / ${ data.quote_token_symbol }` : '';
+
+  return (
+    <>
+      <PageTitle
+        title={ poolTitle }
+        backLink={ backLink }
+        beforeTitle={ data ? (
+          <PoolEntity.Icon
+            pool={ data }
+            isLoading={ isPlaceholderData }
+            size="lg"
+          />
+        ) : null }
+        contentAfter={ <Tag>Pool</Tag> }
```

Comment:
When the page is loading, we should display a skeleton instead of the actual tag.

Reply:
not sure but ok

Reply:
Let's use our `ui/shared/chakra/Tag.tsx` component for this, as it already has an implemented loading state.